spark-datasource/build.gradle.kts

```diff
@@ -0,0 +1,32 @@
+plugins {
+    id("scala")
+    id("java-conventions")
```

Comment:
Should `java-conventions` be here? We should figure out what is needed from that and move it to `scala-conventions`

Reply:
would it rather make sense to move this into a `code-formatting.gradle.kts` file which would cover all code formatting (Java, Scala and "gitattributes")?

Having gitattributes formatting in the Java conventions is weird, anyway...

Is there a way to apply a *.gradle.kts globally, across all sub-projects? Preferably, the code-formatting.gradle.kts would be completely global.